### ISSUE TITLE
Saving an Image to a checklist fails

### ISSUE THREAD

**Issue by DanielCaspers** [DanielCaspers/DigitalInspection#28]
The folders "Uploads/Checklists/" and any similar folders weren't autogenerated on saving a picture. These folders should be 777 perms or higher.

Improve ImageService Implementation to create both directories prior to any save.

**Comment by DanielCaspers** [comment]
Commit 73d25338af2a3638ed7ea699d5afb9e8c455238f should address this. Feel free to take a read of the changes if you like. In deploying to Nightly B, I saw a strange error regarding datetime conversion on fetching the list, which should be an independent action of this, so we may have database corruption issues. 

@scaspers, could you blow away the DB so we can test again? Nightly-B should not have those missing folders and then should make them JIT on image save.

**Comment by scaspers** [comment]
Still having issues.  Failed to create the "Uploads" directory so I manually created with perms 777 and tried again. Then...
![image](https://cloud.githubusercontent.com/assets/12421599/24335456/6fa1d166-1243-11e7-8070-dcfdd8256c01.png)
Upon looking at the 'Uploads' directory, I found that it made the 'Checklists' directory with user 'apache' and perms 700 (read\write to owner only).  The directory did NOT have the image stored in it.  I am guessing that the image did not store due to the program failing before it had a chance.  It probs made the directory and then tried changing perms and failed. ???

**Comment by scaspers** [comment]
Would working with the linux database help?  I could maybe open the firewall for you to the SQL server??

**Comment by DanielCaspers** [comment]
I was worried that would happen.... I'll be investigating what I can do in mono about this.

[Directory.SetAccessControl() documentation](https://msdn.microsoft.com/en-us/library/system.io.directory.setaccesscontrol(v=vs.110).aspx)

While not related to the exact exception above, if I could have access to work with the database directly and view it and operate on it like you did in the browser, that would be a big help not for this bug but for all the other database issues.

**Comment by DanielCaspers** [comment]
[Here's the culprit in Mono's source](https://github.com/mono/mono/blob/master/mcs/class/corlib/System.Security.AccessControl/NativeObjectSecurity.cs#L233)... The stacktrace was actually helpful for a change!

I'll look to see if there is another way around that doesn't take much time for now; otherwise it might be something to revisit and solve in a different way than directly invoking that API and instead maybe run a script that does it or something.

Alternatively, I may start investigation of migrating to .NET Core in the near future, where all of these basics are implemented natively.

**Comment by scaspers** [comment]
Maybe this will help with setting permissions.
[http://docs.go-mono.com/?link=M%3aMono.Unix.Native.Syscall.chmod(System.String%2cMono.Unix.Native.FilePermissions)](http://docs.go-mono.com/?link=M%3aMono.Unix.Native.Syscall.chmod(System.String%2cMono.Unix.Native.FilePermissions))

**Comment by DanielCaspers** [comment]
This is gonna be a time sink, and both of us don't want me spending time on this right now. Lets kick this can a bit further down the road.

@scaspers , let me know which idea you prefer to run with from below. I'll need your help either way.

1) I have folder creation working on every publish. I no longer delete everything on every publish, so we might be able to change folder perms and they may not be affected on new publishes. Not sure though, since perms structure is different across systems. So, could you change Nightly-B's folders to chmod 777.

2) I've already created the folders. Just give apache or whatever access to do whatever it wants for now on those subdirectories using a terminal command? The server isn't crashing anymore, so it should just be once, and only ever redone on server reboots?

**Comment by DanielCaspers** [comment]
@scaspers did I sell you on deferring this? Not trying to be pushy, just trying to manage the backlog!

**Comment by scaspers** [comment]
Yes we can defer.  After we figure out how many checklists we might actually have we may even decide that a list view would be better.

**Comment by DanielCaspers** [comment]
@scaspers , agreed. If you decide you think the list view would be better, I'm thinking I could make that swap in a long evening at worst so if you sense a need to change the UI for that, we can work that out for V1.
